ydb/core/tablet_flat/flat_part_iter_multi.h

```diff
@@ -1142,17 +1147,15 @@ namespace NTable {
         Y_FORCE_INLINE void InitKey() const noexcept
         {
             Y_VERIFY_DEBUG(Main.IsValid(), "Attempt to get an invalid key");
+            if (KeyInitialized)
+                return;
 
-            if (!Key) {
-                auto& info = Part->Scheme->Groups[0].ColsKeyData;
-
-                for (size_t pos = 0; pos < info.size(); ++pos) {
-                    Key.push_back(Main.GetRecord()->Cell(info[pos]));
-                }
+            KeyInitialized = true;
 
-                for (size_t pos = info.size(); pos < KeyCellDefaults->Size(); ++pos) {
-                    Key.push_back((*KeyCellDefaults)[pos]);
-                }
+            auto& info = Part->Scheme->Groups[0].ColsKeyData;
+            size_t infoSize = info.size();
+            for (size_t pos = 0; pos < infoSize; ++pos) {
+                Key[pos] = Main.GetRecord()->Cell(info[pos]);
```

Comment:
Makes me wonder if this could be more efficient:

```
auto it = Key.begin();
auto* record = Main.GetRecord();
for (auto& info : Part->Schema->Groups[0].ColsKeyData) {
    *it++ = record->Cell(info);
}
```

I think iterations using pointers are often more efficient than indexes?

Reply:
Updated.